charts/rstudio-connect/ci/complex-values.yaml

```diff
@@ -25,6 +25,9 @@ service:
 podDisruptionBudget:
   minAvailable: 30%
 
+nodeSelector:
+  something: special
```

Comment:
I'm not sure I understand the purpose of this. Typically, a nodeSelector is used to force (or at least encourage) a pod to be scheduled on specific worker nodes. Does https://github.com/rstudio/helm/blob/main/charts/rstudio-connect/values.yaml#L57-L58 mean that we are _required_ to put something here?

Reply:
Well this ci files was definitely out of date and referencing `pod.nodeSelector` instead of `NodeSelector` which was incorrect, so this change was needed.

> Does https://github.com/rstudio/helm/blob/main/charts/rstudio-connect/values.yaml#L57-L58 mean that we are required to put something here?

If you take a look at https://github.com/rstudio/helm/blob/main/charts/rstudio-connect/templates/deployment.yaml#L43-L46 you'll see where this is referenced. `nodeSelector` will be an empty `{}` list by default, and thus won't even unwrap that template block:

```yaml
      {{- with .Values.nodeSelector }}
      nodeSelector:
        {{- toYaml . | nindent 8 }}
      {{- end }}
```

So you aren't required to set a `nodeSelector` value, if you do set one though that will be translated into a nodeSelector on the connect pod